flutter_mobx/test/flutter_mobx_test.dart

```diff
@@ -35,6 +39,23 @@ class LoggingObserver extends Observer {
   }
 }
 
+// ignore: must_be_immutable
+class FlutterErrorThrowingObserver extends Observer {
+  // ignore: prefer_const_constructors_in_immutables
+  FlutterErrorThrowingObserver({
+    @required WidgetBuilder builder,
+    Key key,
+  }) : super(key: key, builder: builder);
+
+  @override
+  State<Observer> createState() => FlutterErrorThrowingObserverState();
+}
+
+class FlutterErrorThrowingObserverState extends ObserverState {
+  @override
+  void invalidate() => throw FlutterError('setState failed!');
```

Comment:
Curious what will happen if the error is just a regular `Error` (eg: `AssertionError`). I think we should have one test for generic errors.